drools-parser/src/main/antlr4/org/drools/parser/DRL6Expressions.g4

```diff
@@ -697,42 +706,45 @@ explicitGenericInvocation
     ;
 
 nonWildcardTypeArguments
-    :	LESS typeList GREATER
+    :	LT typeList GT
     ;
 
 explicitGenericInvocationSuffix
     :	super_key superSuffix
-    |   	ID arguments
+    |   	IDENTIFIER arguments
     ;
 
 selector
-    :   (DOT super_key)=>DOT { helper.emit($DOT, DroolsEditorType.SYMBOL); } super_key superSuffix
-    |   (DOT new_key)=>DOT { helper.emit($DOT, DroolsEditorType.SYMBOL); } new_key (nonWildcardTypeArguments)? innerCreator
-    |   (DOT ID)=>DOT { helper.emit($DOT, DroolsEditorType.SYMBOL); }
-                  ID { helper.emit($ID, DroolsEditorType.IDENTIFIER); }
-                  ((LEFT_PAREN) => arguments)?
-    |   (NULL_SAFE_DOT ID)=>NULL_SAFE_DOT { helper.emit($NULL_SAFE_DOT, DroolsEditorType.SYMBOL); }
-                  ID { helper.emit($ID, DroolsEditorType.IDENTIFIER); }
-                  ((LEFT_PAREN) => arguments)?
+    :   DOT { helper.emit($DOT, DroolsEditorType.SYMBOL); } super_key superSuffix
+    |   DOT { helper.emit($DOT, DroolsEditorType.SYMBOL); } new_key (nonWildcardTypeArguments)? innerCreator
+    |   DOT { helper.emit($DOT, DroolsEditorType.SYMBOL); }
+                  IDENTIFIER { helper.emit($IDENTIFIER, DroolsEditorType.IDENTIFIER); }
+                  (arguments)?
+    |   NULL_SAFE_DOT { helper.emit($NULL_SAFE_DOT, DroolsEditorType.SYMBOL); }
+                  IDENTIFIER { helper.emit($IDENTIFIER, DroolsEditorType.IDENTIFIER); }
+                  (arguments)?
     //|   DOT this_key
-    |   (LEFT_SQUARE)=>LEFT_SQUARE { helper.emit($LEFT_SQUARE, DroolsEditorType.SYMBOL); }
+    |   LBRACK { helper.emit($LBRACK, DroolsEditorType.SYMBOL); }
                        expression
-                       RIGHT_SQUARE { helper.emit($RIGHT_SQUARE, DroolsEditorType.SYMBOL); }
+                       RBRACK { helper.emit($RBRACK, DroolsEditorType.SYMBOL); }
     ;
 
 superSuffix
     :	arguments
-    |   	DOT ID ((LEFT_PAREN) => arguments)?
+    // TODO syntactic predicates in the form of `(x) => x` can be safely removed but
+    //  there was originally `DOT ID ((LEFT_PAREN) => arguments)?`.
+    //  Not sure if removing `(LEFT_PAREN) =>` is correct.
```

Comment:
No, this is OK because `arguments` can only start with a `LPAREN` so the syn. predicate is indeed redundant.